pkg/driver/driver_azure.go

```diff
@@ -502,48 +503,60 @@ func (d *AzureDriver) createVMNicDisk() (*compute.VirtualMachine, error) {
 	}
 	onARMAPISuccess(prometheusServiceSubnet, "subnet.Get")
 
-	/*
-		NIC creation
-	*/
+	NIC, err := clients.nic.Get(ctx, resourceGroupName, nicName, "")
+	//klog.Infof("Error: %v \nNIC: %v", err, NIC)
 
-	// Creating NICParameters for new NIC creation request
-	NICParameters := d.getNICParameters(vmName, &subnet)
+	// Change to something like below, however I am unable to find the helper method for this.
+	// if err != nil && management.IsResourceNotFoundError(err) {
```

Comment:
I am having some trouble with validating the error code when NIC deosn't exist. Any suggestions? 

cc: @AxiomSamarth , @amshuman-kr

Reply:
https://github.com/Azure/go-autorest/blob/master/autorest/error.go#L41 might be worth a shot.

Reply:
Based on your suggestion i have updated the logic here to this - https://github.com/gardener/machine-controller-manager/pull/594/files#diff-36365e28193d28c39f36ac4d9172fbf8eac69553361f6179fa3184f9cad9eb0cR1329-R1342. I hope this is more appropriate now.